tests/system/test_client.py

```diff
@@ -1052,6 +1052,211 @@ def test_load_table_from_file_w_explicit_location(self):
                 table_ref, "gs://{}/letters-us.csv".format(bucket_name), location="US"
             ).result()
 
+    def test_create_external_table_with_reference_file_schema_uri_avro(self):
+        client = Config.CLIENT
+        dataset_id = _make_dataset_id("external_reference_file_avro")
+        self.temp_dataset(dataset_id)
+        dataset_ref = bigquery.DatasetReference(client.project, dataset_id)
+        table_id = "test_ref_file_avro"
+        table_ref = bigquery.TableReference(dataset_ref=dataset_ref, table_id=table_id)
+
+        expected_schema = [
+            bigquery.SchemaField("username", "STRING", mode="NULLABLE"),
+            bigquery.SchemaField("tweet", "STRING", mode="NULLABLE"),
+            bigquery.SchemaField("timestamp", "STRING", mode="NULLABLE"),
+            bigquery.SchemaField("likes", "INTEGER", mode="NULLABLE"),
+        ]
+
+        # By default, the table should have the c-twitter schema because it is lexicographically last:
+        # a-twitter schema: (username, tweet, timestamp, likes)
+        # b-twitter schema: (username, tweet, timestamp)
+        # c-twitter schema: (username, tweet)
+        source_uris = [
+            "gs://cloud-samples-data/bigquery/federated-formats-reference-file-schema/a-twitter.avro",
+            "gs://cloud-samples-data/bigquery/federated-formats-reference-file-schema/b-twitter.avro",
+            "gs://cloud-samples-data/bigquery/federated-formats-reference-file-schema/c-twitter.avro",
+        ]
+
+        # Because referenceFileSchemaUri is set as a-twitter, the table will have a-twitter schema
+        reference_file_schema_uri = "gs://cloud-samples-data/bigquery/federated-formats-reference-file-schema/a-twitter.avro"
```

Comment:
non blocking nit)- for maintainability, is it worth setting these to variables or constants and then using the same variable in `source_uris` and `reference_file_schema_uri`?

Reply:
Good catch, I think it's worth it